layer4/listener.go

```diff
@@ -33,7 +33,7 @@ type ListenerWrapper struct {
 }
 
 // CaddyModule returns the Caddy module information.
-func (ListenerWrapper) CaddyModule() caddy.ModuleInfo {
+func (*ListenerWrapper) CaddyModule() caddy.ModuleInfo {
```

Comment:
I haven't finished my review yet but does this actually solve a problem? I _guess_ I don't care either way in the cases where this isn't even used (there's not even a variable named for it) but I guess I just am curious.